.github/workflows/fast-forward-pr-merge-init.md

```diff
@@ -0,0 +1,72 @@
+
+If the user passes these checks, a second workflow, [`fast-forward-pr-merge-privileged.yml`](./fast-forward-pr-merge-privileged.md), is triggered to perform the actual fast-forward merge and post the final result.
+
+It is especially important when accepting contributions from external users via a forked repository, which is how most open source contributions are made.
+
+## Why is the Workflow Split into Two Phases?
+
+Splitting the merge process into two distinct workflows is a best practice for securing GitHub Actions, especially when privileged operations (like merging to a protected branch) are involved. This approach is recommended by GitHub Security Lab and is designed to mitigate several classes of vulnerabilities:
+
+### Security Boundaries
+
+By separating the unprivileged (label-triggered) phase from the privileged (merge-executing) phase, you create a strong security boundary. The first workflow runs with minimal permissions and only performs checks and artifact publishing. The second, privileged workflow (triggered by `workflow_run`) runs with elevated permissions and only after all checks have passed.
+
+### Prevents Privilege Escalation
+
+If a single workflow both checked permissions and performed the merge, an attacker could potentially exploit timing or event confusion such as **Time Of Check to Time Of Use** ([TOCTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use)) attacks to escalate privileges or inject malicious code between steps. By splitting the workflows, the privileged phase only runs after the unprivileged phase has completed and its outputs (artifacts) are validated.
+
+### Mitigates Artifact Poisoning
+
+Artifacts passed from the unprivileged to the privileged workflow are treated as untrusted. This separation allows you to validate and sanitise any data before it is used in a privileged context, reducing the risk of artifact poisoning attacks.
+
+### Reduces Risk from Untrusted Triggers
+
+Triggers like `issue_comment` or `pull_request_target` can be abused by attackers to execute workflows with elevated permissions. By using a label gate and a two-phase approach, you ensure that only trusted actors (with `write` or `admin` permissions) can initiate the privileged merge, and that the code being merged is exactly what was reviewed and approved.
+
+### Aligns with GitHub Security Guidance
+
+GitHub’s own security research recommends splitting workflows into unprivileged and privileged components, using `workflow_run` as a secure handoff point. This pattern is now widely adopted in the open source community to prevent supply chain attacks and workflow abuse.
+
+
+## When Does This Workflow Run?
+
+- **Trigger:** The workflow is triggered when a pull request receives the `fast-forward` label (`pull_request` event, type `labeled`).
+- **Conditions:**
+  - The event must be a pull request being labelled.
+  - The label must be `fast-forward`.
+  - The PR must be open (not closed).
```

Comment:
Suggest removing. I'm not sure this adds much, since this detail is clear from the code. And I worry that it will drift.